src/schema/schema_create.c

```diff
@@ -433,6 +433,70 @@ __wt_find_import_metadata(WT_SESSION_IMPL *session, const char *uri, const char
     return (0);
 }
 
+/*
+ * __schema_is_tiered_storage_shared --
+ *     Check whether the table is configured for tiered storage, and if so, whether the tiered table
+ *     is shared.
+ */
+static bool
+__schema_is_tiered_storage_shared(WT_SESSION_IMPL *session, const char *config)
+{
+    WT_CONFIG_ITEM cval;
+
+    /*
+     * The tiered storage shared table needs to have two column groups that are
+     * pointed to the underlying active and shared files. The following checks are
+     * carried out to determine the table can be created as a tiered storage shared
+     * or not based on the table creation configuration where they control the
+     * underlying the source.
+     *
+     * 1. The table configuration should not specify underlying source.
+     * 2. The table configuration should not specify underlying type of the storage.
+     * 3. The connection is not configured for tiered storage or the table configured
+     *    for non tiered storage.
+     * 4. The connection is not configured for tiered storage shared or the table is
+     *    configured for non tiered storage shared.
```

Comment:
Same here:
4. The connection is not configured for tiered storage or the table is not configured for shared tiered storage.

Reply:
Thanks. Fixed.